### PR TITLE
Move to react #82

### PR DESCRIPTION
**Pull request by balthazzar**
https://github.com/webRunes/Core-WRIO-App/issues/82

### CODE REVIEW

**Comment by AlexeyAnshakov** [comment]
@michbil please review

File: src/client/js/mentions/mention.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+var sortBy = require('lodash.sortby');
```
Comment (michbil) [inline src/client/js/mentions/mention.js]: @balthazzar Use ES6 classes, Luke

File: src/client/js/client.js [hunk 2]
```diff
@@ -1 +1,500 @@
...
+		if (!json) return;
+
+		var textToWrite = this.state.cleshe.replace('|BODY|', JSON.stringify(json));
+		textToWrite = textToWrite.replace('|TITLE|', json.name);
+
+		//ToDo: test
+		// /console.log(textToWrite);
+		var url = "zope.html";
+		var contents = "<html></html>";
+
```
Comment (michbil) [inline src/client/js/client.js]: i think above two lines not needed now

**Comment by balthazzar** [comment]
@michbil please review

**Comment by michbil** [comment]
@AlexeyAnshakov seems good to me

File: src/client/js/client.js [hunk 3]
```diff
@@ -1 +1,495 @@
...
+		textToWrite = textToWrite.replace('|TITLE|', json.name);
+
+
+
+		//ToDo: test
+		//console.log(textToWrite);
+		var url = this.state.saveUrl;
+		var contents = "<html></html>";
+		$.ajax({
+				url: "http://storage."+domain+"/api/save",
```
Comment (AlexeyAnshakov) [inline src/client/js/client.js]: must be https

Comment (michbil) [inline src/client/js/client.js]: @balthazzar @AlexeyAnshakov 
Тут надо подумать, т.к. если насильно включтить https поломается local development
Скорее всего надо написать метод определения домена и использовать именно его

```
function getStorageDomain() {
	var protocol = 'https://';
	if (domain === 'wrioos.local') {
		protocol = 'http://';
	}
	return protocol +"storage."+domain+"/"
}
```

File: src/client/js/mentions/scripts.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		json,
+		data = [];
+	for (i = 0; i < scripts.length; i += 1) {
+		if (scripts[i].type === 'application/ld+json') {
+			json = undefined;
+			try {
+				json = JSON.parse(scripts[i].textContent);
+			} catch (exception) {
+				json = undefined;
+				console.error('Your json-ld not valid: ' + exception);
```
Comment (AlexeyAnshakov) [inline src/client/js/mentions/scripts.js]: Invalid JSON-LD

File: src/client/views/core.htm [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="description" content="Official webRunes WRIO OS node">
+    <meta name="author" content="webRunes">
+    <script src="//code.jquery.com/jquery-1.11.2.min.js"></script>
+    <link href="http://webrunes.github.io/Default-WRIO-Theme/css/bootstrap.css" rel="stylesheet" />
```
Comment (AlexeyAnshakov) [inline src/client/views/core.htm]: https://webrunes.github.io/Default-WRIO-Theme/css/bootstrap.css

File: src/client/views/core.htm [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    <link href="http://webrunes.github.io/Default-WRIO-Theme/css/webrunes.css" rel="stylesheet">
```
Comment (AlexeyAnshakov) [inline src/client/views/core.htm]: https://webrunes.github.io/Default-WRIO-Theme/css/webrunes.css

File: src/client/js/client.js [hunk 7]
```diff
@@ -1 +1,495 @@
...
+				}
+			})
+			.success((res) => {
+				window.location = res.url;
+			});
+	};
+
+	disableSave() {
+		$('#save-button-id div a').addClass('disabled');
+		$('#save-button-id div').attr('title', 'You can\'t save to non webrunes hosted page. Use \"Save As\" and upload file manually');
```
Comment (AlexeyAnshakov) [inline src/client/js/client.js]: You can only save the page at webrunes.com. Use "Save as" and upload file manually.

File: src/client/js/client.js [hunk 8]
```diff
@@ -1 +1,495 @@
...
+
+		var fileName = (json.name === '' ? 'untitled' : json.name.split(' ').join('_')) + '.htm';
+		if (ie || ie11 || ieEDGE) {
+			if (ieVer>9 || ieEDGE) {
+				var textFileAsBlob = new Blob([textToWrite], {
+					type: 'text/plain'
+				});
+				window.navigator.msSaveBlob(textFileAsBlob, fileName);
+			} else {
+				console.log("No supported on IE ver<10");
```
Comment (AlexeyAnshakov) [inline src/client/js/client.js]: IE v.10 or higher required

File: src/client/js/client.js [hunk 9]
```diff
@@ -1 +1,495 @@
...
+		}
+	}
+
+	parseEditingUrl() {
+		var editUrl = window.location.search.match(/\?article=([\.0-9a-zA-Z%:\/?]*)/);
+		if (editUrl) {
+			this.state.editUrl = editUrl[1];
+			editUrl = this.state.editUrl;
+			var editUrlParsed = this.getLocation(editUrl);
+			console.log("Got editing url", editUrl);
```
Comment (AlexeyAnshakov) [inline src/client/js/client.js]: Page edit link received

File: src/index.js [hunk 10]
```diff
@@ -29,18 +29,18 @@ var mongoUrl = 'mongodb://' + nconf.get('mongo:user') + ':' + nconf.get('mongo:p
 db.mongo({
 		url: mongoUrl
 	})
-	.then(function(res) {
+	.then((res) => {
 		console.log("Connected correctly to database");
```
Comment (AlexeyAnshakov) [inline src/index.js]: A connection was successfully established with the server

File: src/index.js [hunk 11]
```diff
@@ -29,18 +29,18 @@ var mongoUrl = 'mongodb://' + nconf.get('mongo:user') + ':' + nconf.get('mongo:p
...
-			.listen(nconf.get("server:port"), function(req, res) {
+			.listen(nconf.get("server:port"), (req, res) => {
 				console.log('app listening on port ' + nconf.get('server:port') + '...');
 				server_setup(db);
 				console.log("Application Started!");
 			});
 	})
-	.catch(function(err) {
+	.catch((err) => {
 		console.log('Error connect to database:' + err.code + ': ' + err.message);
```
Comment (AlexeyAnshakov) [inline src/index.js]: Error connecting to DataBase

File: src/index.js [hunk 12]
```diff
@@ -61,28 +61,18 @@ function server_setup(db) {
...
-		});
+	app.get('/create', (request, response) => {
+		response.sendFile(__dirname +
+			'/client/views/core.htm');
 	});
 
-	app.get('/edit', function(request, response) {
-		getArticle(request.query.article, function(err, article) {
-			if (err) {
-				console.log("Got error: " + err.message);
```
Comment (AlexeyAnshakov) [inline src/index.js]: Error:

File: src/client/js/client.js (hunk 3 above)
Comment (AlexeyAnshakov) [inline src/client/js/client.js]: In this case let be //

File: src/client/views/core.htm [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en-US">
+
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta name="description" content="Official webRunes WRIO OS node">
```
Comment (AlexeyAnshakov) [inline src/client/views/core.htm]: Core editor

**Comment by balthazzar** [comment]
@michbil @AlexeyAnshakov fixed

**Pull request merged**